File: 0179-largest-number/0179-largest-number.cs
public class Solution {
    public string LargestNumber(int[] nums) {
        string[] strNums = new string[nums.Length];
        for (int i = 0; i < nums.Length; i++) 
            strNums[i] = nums[i].ToString();
        Array.Sort(strNums, (x, y) => (y + x).CompareTo(x + y));
        string res = string.Concat(strNums);
        return res[0] == '0' ? "0" : res;
    }
}

File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> strNums(nums.size());
        for (int i = 0; i < nums.size(); ++i) strNums[i] = to_string(nums[i]);
        sort(strNums.begin(), strNums.end(), [](const string& a, const string& b) {
            return a + b > b + a;
        });
        string res = accumulate(strNums.begin(), strNums.end(), string());
        return res[0] == '0' ? "0" : res;
    }
};
